src/data-sources/DataSourceList.tsx

```diff
@@ -152,8 +149,19 @@ const DataSourceList = () => {
 		table: {},
 	};
 
-	const isItemEligibleForActions = ( item: DataSourceConfig ) => {
-		return item.config_source === ConfigSource.STORAGE;
+	const isItemEligibleForActions = ( item: DataSourceConfig, action: string ) => {
+		if ( item.config_source !== ConfigSource.STORAGE ) {
+			return false;
+		}
+
+		// This is meant to limit actions on disabled items. For an item to be disabled, it must have the enabled field set to false.
+		// The only actions allowed on disabled items are delete and copy as they won't cause any unintended side effects.
+		const actionsAllowedForDisabledItems = [ 'delete', 'copy' ];
```

Comment:
This is the new check on the UI that limits the operations on disabled items. Note that I am explicitly looking for the enabled flag to be set to false, so that in case a data source is given back without it it won't lead to bugs (that shouldn't happen but this catches it).

I've also added some explanation for the logic here as well.